Autonomous_Parallel_Parking/Autonomous_Parallel_Parking.c

```diff
@@ -15,14 +15,84 @@
  ****************************************************************************************************************/
 
 #include "cpu.h"
-#include "board_led.h"
+#include "pin_setup.h"
 
-#include "foo.h"
+int count = 0;
 
 int main()
 {
   cpu_init();
-  board_led_init();
-
+  GPIOInterruptSetup();
   return 0;
 }
+
+/**
+* @brief IRQ for PA0 (user button) - to test if PA0 interrupt is functioning.  Turns on blue LED every 5 times the 
+* button is pressed.
+*/
+void EXTI0_IRQHandler(void){
```

Comment:
Yes i completely agree; however, i do think that things such as pin initialization should come before the while(1) loop. so it would be something like:

``` C
int main()
{
     cpu_init();
     GPIOInterruptSetup();
     //any other inits
     while(1)
     {
           //state machine here
     }
     return 1;
}
```

And then have the definition of `GPIOInterruptSetup` elsewhere, correct?